sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionImplUtils.java

```diff
@@ -62,13 +62,13 @@ private static byte[] aesInternal(
       }
 
     try {
-      if (mode.equalsIgnoreCase("ECB") &&
-          (padding.equalsIgnoreCase("PKCS") || padding.equalsIgnoreCase("DEFAULT"))) {
+      if ("ECB".equalsIgnoreCase(mode) &&
+          ("PKCS".equalsIgnoreCase(padding) || "DEFAULT".equalsIgnoreCase(padding))) {
         Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
         cipher.init(opmode, secretKey);
         return cipher.doFinal(input, 0, input.length);
-      } else if (mode.equalsIgnoreCase("GCM") &&
-          (padding.equalsIgnoreCase("NONE") || padding.equalsIgnoreCase("DEFAULT"))) {
+      } else if ("GCM".equalsIgnoreCase(mode) &&
+          ("NONE".equalsIgnoreCase(padding) || "DEFAULT".equalsIgnoreCase(padding))) {
```

Comment:
Could you revert these changes of this file, @dcoliversun ?
This is logically impossible because `aesInternal` is `private static` and used in the constructor with `mode.toString()`.